src/jit/function-builder.h

```diff
@@ -29,79 +30,63 @@
 namespace wabt {
 namespace jit {
 
+class WabtState : public TR::VirtualMachineState {
+public:
+  VirtualStack stack;
+
+  virtual VirtualMachineState* MakeCopy() {
+    return new WabtState(*this);
+  }
+
+  virtual void MergeInto(TR::VirtualMachineState* other, TR::IlBuilder* b) {
+    this->stack.MergeInto(&static_cast<WabtState*>(other)->stack, b);
```

Comment:
Just switching from `static_cast<WabtState*>` to `dynamic_cast<WabtState*>` wouldn't actually gain any safety, since `dynamic_cast` of a pointer type returns `nullptr` if the cast is invalid. Since dereferencing `nullptr` is UB, we wouldn't actually gain anything.

I could replace this with `&dynamic_cast<WabtState&>(*other).stack` though... Thoughts?

Reply:
I like `&dynamic_cast<WabtState&>(*other).stack`. I really just want to have something "defined" happen if the cast becomes invalid because of some change we do later on to the class hierarchy :slightly_smiling_face:.